Tools/clinic/clinic.py

```diff
@@ -1384,7 +1393,10 @@ def parser_body(
                 )
                 nargs = f"Py_MIN(nargs, {max_pos})" if max_pos else "0"
 
-            if clinic.limited_capi:
+            limited_capi = clinic.limited_capi
+            if requires_defining_class or pseudo_args:
+                limited_capi = False
```

Comment:
Consider producing a warning if we fall back to `limited_capi = False`.

Reply:
Done. Also it now detects more incompatible input and no longer needs to exclude specific files by name.